lib/Transforms/InstCombine/InstCombineCasts.cpp

```diff
@@ -1447,6 +1447,11 @@ Instruction *InstCombiner::commonPointerCastTransforms(CastInst &CI) {
     // If casting the result of a getelementptr instruction with no offset, turn
     // this into a cast of the original pointer!
     if (GEP->hasAllZeroIndices() &&
+        // HLSL Change Begin.
+        // Ignore this optimization for pointer bitcast to pass validation.
+        // TODO: enable this optimization when validation is updated.
+        0 &&
```

Comment:
Why are we just disabling this? If the validator is complaining then the right fix is to update the validator or fix the codegen so it is happy.